src/cudamatrix/cu-kernels.cu

```diff
@@ -2217,6 +2217,66 @@ static void _softmax_reduce(Real*y, const Real*x, MatrixDim d, int src_stride) {
   }
 }
 
+
```

Comment:
Great work, but would you mind adding a comment explaining what this does, and some other details about it?